### PR TITLE
dialect/sql/schema: add support for index prefixes in mysql

### PR DESCRIPTION
**Pull request by a8m**
Fixed https://github.com/ent/ent/issues/1193

### CODE REVIEW

File: dialect/entsql/annotation.go [hunk 1]
```diff
@@ -158,3 +158,100 @@ const (
...
+	}
+}
+
+// Name describes the annotation name.
+func (IndexAnnotation) Name() string {
+	return "EntSQLIndexes"
+}
+
+// Merge implements the schema.Merger interface.
+func (a IndexAnnotation) Merge(other schema.Annotation) schema.Annotation {
```
Comment (rotemtam) [inline dialect/entsql/annotation.go:228]: 💯

File: entc/integration/migrate/entv1/migrate/schema.go [hunk 2]
```diff
@@ -86,22 +86,30 @@ var (
...
 				OnDelete:   schema.SetNull,
 			},
 		},
 		Indexes: []*schema.Index{
+			{
+				Name:    "user_description",
+				Unique:  false,
+				Columns: []*schema.Column{UsersColumns[3]},
+				Annotation: &entsql.IndexAnnotation{
+					Prefix: 50,
```
Comment (rotemtam) [inline entc/integration/migrate/entv1/migrate/schema.go:106]: 💯

**Review by rotemtam — APPROVED** [review]
LGTM well done

**Comment by a8m** [comment]
cc @giautm

**Pull request merged**